index.html

```diff
@@ -204,55 +204,1951 @@ <h3>
+          </td>
+          <td>
+            An [=ordered set=] of {{HIDDevice}} instances representing the
+            [=HID interfaces=] currently available on the system.
+          </td>
+        </tr>
+      </table>
+      <h3>
+        <dfn>hid</dfn> attribute
+      </h3>
+      <p>
+        When getting, the {{Navigator/hid}} attribute always returns the same
+        instance of the {{HID}} object.
+      </p>
+    </section>
+    <section data-dfn-for="HID">
+      <h2>
+        `HID` interface
+      </h2>
+      <pre class="idl">
+        [Exposed=Window, SecureContext]
         interface HID : EventTarget {
             attribute EventHandler onconnect;
             attribute EventHandler ondisconnect;
             Promise&lt;sequence&lt;HIDDevice&gt;&gt; getDevices();
             Promise&lt;sequence&lt;HIDDevice&gt;&gt; requestDevice(
                 HIDDeviceRequestOptions options);
         };
-
-        [SecureContext] partial interface Navigator {
-            [SameObject] readonly attribute HID hid;
+      </pre>
+      <dl>
+        <dt>
+          <dfn>onconnect</dfn> attribute
+        </dt>
+        <dd>
+          {{HID/onconnect}} is an [=event handler IDL attribute=] for the
+          <dfn>connect</dfn> event type.
+        </dd>
+        <dt>
+          <dfn>ondisconnect</dfn> attribute
+        </dt>
+        <dd>
+          {{HID/ondisconnect}} is an [=event handler IDL attribute=] for the
+          <dfn>disconnect</dfn> event type.
+        </dd>
+      </dl>
+      <p>
+        When a [=HID interface=] becomes available on the system, run the
+        following steps:
+      </p>
+      <div class="note">
+        <p>
+          A <dfn>HID interface</dfn> is logical interface that can be used to
+          communicate with a device over the HID protocol. Each [=HID
+          interface=] has exactly one [=Report descriptor=] that describes the
+          set of reports supported by that interface. A single physical device
+          may expose more than one [=HID interface=]. Devices with multiple
+          [=HID interfaces=] run these steps once for each interface.
+        </p>
+      </div>
+      <ol>
+        <li>Let |device:HIDDevice| be a {{HIDDevice}}.
+        </li>
+        <li>Set |device|.{{HIDDevice/vendorId}} to the [=vendor ID=] of the
+        device that exposes the interface.
+        </li>
+        <li>Set |device|.{{HIDDevice/productId}} to the [=product ID=] of the
+        device that exposes the interface.
+        </li>
+        <li>Set |device|.{{HIDDevice/productName}} to the [=product name=] of
+        the device that exposes the interface.
+        </li>
+        <li>Set |device|.{{HIDDevice/collections}} to the result of [=parsing
+        the Report descriptor=].
+        </li>
+        <li>Append |device| to |navigator|.{{Navigator/[[hidDevices]]}}.
+        </li>
+        <li>If the user has allowed the site to access |device| as the result
```

Comment:
I think it's still not 100% specified since I don't have a good way to access |hid| from the "becomes available" and "becomes unavailable" algorithms. I'm referencing [=this=] but these algorithms aren't defined on an object.

Reply:
Move step 1 down to before step 7 and replace "this" with "device".